File: 04_quicksort/ruby/04_recursive_max.rb
def max(list)
  if list.length == 2
    # condition ? then : else
    list[0] > list[1] ? list[0] : list[1]
  else
    sub_max = max(list[1..-1])
    list[0] > sub_max ? list[0] : sub_max
  end
end

File: 04_quicksort/golang/04_recursive_max.go
package main

import "fmt"

func max(list []int) int {
	if len(list) == 2 {
		if list[0] > list[1] {
			return list[0]
		}
		return list[1]
	}

	subMax := max(list[1:])
	if list[0] > subMax {
		return list[0]
	}
	return subMax
}

func main() {
	fmt.Println(max([]int{1, 5, 10, 25, 16, 1}))
}
